src/chains.jl

```diff
@@ -333,6 +333,94 @@ Base.last(c::Chains) = last(c.value[Axis{:iter}].val)
 
 Base.convert(::Type{Array}, chn::Chains) = convert(Array, chn.value)
 
+# Convenience functions to handle different types of 
+# timestamps.
+to_datetime_vec(ts::DateTime) = [ts]
+to_datetime_vec(ts::Vector{DateTime}) = ts
+to_datetime_vec(ts::Float64) = [unix2datetime(ts)]
+to_datetime_vec(ts::Vector{Float64}) = unix2datetime.(ts)
+function to_datetime_vec(ts)
+    @warn "Timestamp type $(typeof(ts)) unknown."
+    return missing
+end
+
+"""
+    start_time(c::Chains)
+
+Retrieve a vector of start times (as `DateTime`)
+from `chain.info`. Assumes that either
+`DateTime` or unix timestamps (`Float64`) are stored in
+`chain.info.start_time`
+"""
+function start_time(c::Chains)
+    if :start_time in keys(c.info)
+        # We've got some times, print them out.
+        ts = c.info.start_time
+        return to_datetime_vec(ts)
+    else
+        # Times not found -- spit out missing.
+        return missing
+    end
+end
+
+function min_start(c::Chains)
+    ts = start_time(c)
+
+    if ts === missing
+        return missing
+    else
+        return minimum(ts)
+    end
+end
+
+"""
+    stop_time(c::Chains)
+
+Retrieve a vector of stop times (as `DateTime`)
+from `chain.info`. Assumes that either
+`DateTime` or unix timestamps (`Float64`) are stored in
+`chain.info.stop_time`
+"""
+function stop_time(c::Chains)
+    if :stop_time in keys(c.info)
+        # We've got some times, print them out.
+        ts = c.info.stop_time
+        return to_datetime_vec(ts)
+    else
+        # Times not found -- spit out missing.
+        return missing
+    end
+end
+
+function max_stop(c::Chains)
+    ts = stop_time(c)
+
+    if ts === missing
+        return missing
+    else
+        return maximum(ts)
+    end
+end
+
+"""
+    wall_duration(c::Chains)
+
+Calculate the wall clock time for all chains in seconds.
+The duration is calculated as the latest stopping time
+minus the earliest starting time.
+"""
+function wall_duration(c::Chains)
+    earliest_start = min_start(c)
+    latest_stop = max_stop(c)
+
+    # DateTime - DateTime returns a Millisecond value,
+    # divide by 1k to get seconds.
+    if earliest_start === missing || latest_stop === missing
+        return missing
+    else
+        return Dates.value(latest_stop - earliest_start) / 1000
+    end
+end
```

Comment:
I suggest that one adds an option to provide the earliest start and the latest stop, similar to e.g. `mean` in `var`, to avoid repeated computations below:
```suggestion
"""
    wall_duration(c::Chains; start=min_start(c), stop=max_stop(c))

Calculate the wall clock time for all chains in seconds.

The duration is calculated as `stop - start`, where as default `stop`
is the latest stopping time and `start` is the earliest starting time.
"""
function wall_duration(c::Chains; start=min_start(c), stop=max_stop(c))
    # DateTime - DateTime returns a Millisecond value,
    # divide by 1k to get seconds.
    return if start === missing || stop === missing
        missing
    else
        Dates.value(stop - start) / 1000
    end
end
```